dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ThreadCreation.java

```diff
@@ -177,9 +180,15 @@ private void handlePthreadJoins(Thread thread, Map<IValue, Expression> tid2ComAd
         final ExpressionFactory expressions = ExpressionFactory.getInstance();
         int joinCounter = 0;
 
-        for (DirectFunctionCall call : thread.getEvents(DirectFunctionCall.class)) {
-            final String targetName = call.getCallTarget().getName();
-            if (!(targetName.equals("pthread_join") || targetName.equals("__pthread_join"))) {
+        for (FunctionCall call : thread.getEvents(FunctionCall.class)) {
+            if (!call.isDirectCall()) {
+                continue;
+            }
+            final String targetName = call.getCalledFunction().getName();
+            if (!(targetName.equals("pthread_join")
+                    // The following two calls can be generated on macOS
+                    || targetName.equals("__pthread_join")
+                    || targetName.equals("\"\\01_pthread_join\""))) {
```

Comment:
can you somehow use the intrinsic Info that groups them all together?

Reply:
Yeah, I guess that should be doable.